src/rust/engine/graph/src/lib.rs

```diff
@@ -1075,12 +1075,13 @@ impl<N: Node> Graph<N> {
     }
 
     // Otherwise, clear the deps.
-    let dep_edges: Vec<_> = inner
+    // NB: Because `remove_edge` changes EdgeIndex values, we remove edges one at a time.
```

Comment:
Would it be useful to look into [`StableGraph`](https://docs.rs/petgraph/0.4.12/petgraph/stable_graph/struct.StableGraph.html) at some point? It seems to be specifically designed to address this concern (although it notes that feature parity is not quite there yet).

Reply:
We recently stopped using StableGraph for efficiency reasons.

Reply:
Great! I will try to `git grep` for these questions in the future.